config.go

```diff
@@ -14,52 +14,61 @@ type Config struct {
+	ExtendTags                   []string `yaml:"extend_tags"`
+	FalconerAddress              string   `yaml:"falconer_address"`
+	Features                     struct {
+		MigrateMetricSinks string `yaml:"migrate_metric_sinks"`
+		MigrateSpanSinks   string `yaml:"migrate_span_sinks"`
+	}
+	FlushFile                    string   `yaml:"flush_file"`
+	FlushMaxPerBody              int      `yaml:"flush_max_per_body"`
+	FlushWatchdogMissedFlushes   int      `yaml:"flush_watchdog_missed_flushes"`
+	ForwardAddress               string   `yaml:"forward_address"`
+	ForwardUseGrpc               bool     `yaml:"forward_use_grpc"`
+	GrpcAddress                  string   `yaml:"grpc_address"`
+	GrpcListenAddresses          []string `yaml:"grpc_listen_addresses"`
+	Hostname                     string   `yaml:"hostname"`
+	HTTPAddress                  string   `yaml:"http_address"`
+	HTTPQuit                     bool     `yaml:"http_quit"`
+	IndicatorSpanTimerName       string   `yaml:"indicator_span_timer_name"`
+	Interval                     string   `yaml:"interval"`
+	KafkaBroker                  string   `yaml:"kafka_broker"`
+	KafkaCheckTopic              string   `yaml:"kafka_check_topic"`
+	KafkaEventTopic              string   `yaml:"kafka_event_topic"`
+	KafkaMetricBufferBytes       int      `yaml:"kafka_metric_buffer_bytes"`
+	KafkaMetricBufferFrequency   string   `yaml:"kafka_metric_buffer_frequency"`
+	KafkaMetricBufferMessages    int      `yaml:"kafka_metric_buffer_messages"`
+	KafkaMetricRequireAcks       string   `yaml:"kafka_metric_require_acks"`
+	KafkaMetricTopic             string   `yaml:"kafka_metric_topic"`
+	KafkaPartitioner             string   `yaml:"kafka_partitioner"`
+	KafkaRetryMax                int      `yaml:"kafka_retry_max"`
+	KafkaSpanBufferBytes         int      `yaml:"kafka_span_buffer_bytes"`
+	KafkaSpanBufferFrequency     string   `yaml:"kafka_span_buffer_frequency"`
+	KafkaSpanBufferMesages       int      `yaml:"kafka_span_buffer_mesages"`
+	KafkaSpanRequireAcks         string   `yaml:"kafka_span_require_acks"`
+	KafkaSpanSampleRatePercent   float64  `yaml:"kafka_span_sample_rate_percent"`
+	KafkaSpanSampleTag           string   `yaml:"kafka_span_sample_tag"`
+	KafkaSpanSerializationFormat string   `yaml:"kafka_span_serialization_format"`
+	KafkaSpanTopic               string   `yaml:"kafka_span_topic"`
+	LightstepAccessToken         string   `yaml:"lightstep_access_token"`
+	LightstepCollectorHost       string   `yaml:"lightstep_collector_host"`
+	LightstepMaximumSpans        int      `yaml:"lightstep_maximum_spans"`
+	LightstepNumClients          int      `yaml:"lightstep_num_clients"`
+	LightstepReconnectPeriod     string   `yaml:"lightstep_reconnect_period"`
+	MetricMaxLength              int      `yaml:"metric_max_length"`
+	MetricSinks                  []struct {
+		Kind   string      `yaml:"kind"`
+		Name   string      `yaml:"name"`
+		Config interface{} `yaml:"config"`
```

Comment:
Wanted to confirm, the idea here is that the sink receiving the configuration would include it's own Config struct and more or less do:
```
realConfig := cfg.(MyRealConfigStruct)
```

Reply:
Roughly yes, but a simple cast will not work though. The `Config interface{}` will be a `map[string]interface{}`, which will need to be unpacked into the struct, probably via a library such as [mapstructure](https://github.com/mitchellh/mapstructure).